lib/js/stacks.js

```diff
@@ -0,0 +1,60 @@
+
+(function () {
+    var application = Stimulus.Application.start();
+
+    window.Stacks = window.Stacks || Object.create(null);
+    Stacks.controllers = Object.create(null);
+
+    function StacksController () {
+        Stimulus.Controller.apply(this, arguments);
+    }
+    StacksController.prototype = Object.create(Stimulus.Controller.prototype, {
+        constructor: { value: StacksController, enumerable: false, writable: true }
+    });
+    StacksController.prototype.getElementData = function(element, key) {
+        return element.getAttribute("data-" + this.identifier + "-" + key);
```

Comment:
FWIW, there's a "private" function, `this.data.getFormattedKey` that does the same conversion, only it'll also desnake key (eg. `fooId -> data-s-table-foo-id`).  Really, it might be worth a PR to Stimulus to add an optional element parameter to `this.data.get` so you could do `this.data.get('some-attribute', this.someTarget)` instead of ` this.getElementData(this.someTarget, 'some-attribute')`

Reply:
I want to play with it a bit whether this way of doing things even makes sense. The Basecamp people are very careful with their choice of what APIs they add (which I appreciate), so maybe this won't even turn out to be a good idea in practice. We'll see :)